Homework2/Program.cs

```diff
@@ -0,0 +1,171 @@
+
+            // Add item to end
+            Console.WriteLine($"After adding to the end: {string.Join(" ", AddToEnd(originalArray, -100))}");
+
+            // Add item at any index
+            int inserIndex = -1;
+            while (inserIndex < 0 || inserIndex >= originalArray.Length)
+            {
+                Console.Write($"Enter an integer between 0 and {originalArray.Length - 1} (inclusive): ");
+                inserIndex = int.Parse(Console.ReadLine());
+            }
+            Console.WriteLine($"After inserting at {inserIndex}: {string.Join(" ", AddToArray(originalArray, -100, inserIndex))}");
+
+            // Remove from the start
+            Console.WriteLine($"After removing from the start: {string.Join(" ", RemoveFromStart(originalArray))}");
+
+            // Remove from the end
+            Console.WriteLine($"After removing from the end: {string.Join(" ", RemoveFromEnd(originalArray))}");
+
+            // Remove from any index
+            int removeIndex = -1;
+            while (removeIndex < 0 || removeIndex >= originalArray.Length)
+            {
+                Console.Write($"Enter an integer between 0 and {originalArray.Length - 1} (inclusive): ");
+                removeIndex = int.Parse(Console.ReadLine());
+            }
+            Console.WriteLine($"After removing from {removeIndex}: {string.Join(" ", RemoveFromArray(originalArray, removeIndex))}");
+
+            SignUp();
+            
+        }
+
+        static int[] SortArray(int[] arr)
+        {
+            int[] copiedArray = new int[arr.Length];
+            Array.Copy(arr, copiedArray, arr.Length);
+            for (int i = 0; i < copiedArray.Length - 1; i++)
+            {
+                int j;
+                int minimum = copiedArray[i];
+                int minIndex = i;
+                for (j = i + 1; j < copiedArray.Length; j++)
+                {
+                    if (copiedArray[j] < minimum)
+                    {
+                        minimum = copiedArray[j];
+                        minIndex = j;
+                    }
+                }
+                int temp = copiedArray[i];
+                copiedArray[i] = minimum;
+                copiedArray[minIndex] = temp;
+            }
+            return copiedArray;
+        }
+
+        static int[] AddToStart(int[] arr, int item)
+        {
+            int[] copiedArray = new int[arr.Length + 1];
+            Array.Copy(arr, 0, copiedArray, 1, arr.Length);
+            copiedArray[0] = item;
+            return copiedArray;
+        }
+
+        static int[] AddToEnd(int[] arr, int item)
+        {
+            int[] copiedArray = new int[arr.Length + 1];
+            Array.Copy(arr, copiedArray, arr.Length);
+            copiedArray[copiedArray.Length - 1] = item;
+            return copiedArray;
+        }
+
+        static int[] AddToArray(int[] arr, int item, int index)
+        {
+            if (index == 0) return AddToStart(arr, item);
```

Comment:
This is backwards.  There is not much advantage to deferring to `AddToStart`/`AddToEnd`.  If you wanted to simplify things you could do the reverse.  `AddToStart` could be `AddToArray(arr, item, 0)` and `AddToEnd` could be `AddToArray(arr, item, arr.Length)`